crates/wit-component/src/validation.rs

```diff
@@ -114,6 +117,26 @@ pub enum ImportInstance {
     Names(IndexMap<String, Import>),
 }
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct PayloadInfo {
+    pub name: String,
+    pub ty: TypeId,
+    pub function: Function,
+    pub key: WorldKey,
+    pub interface: Option<InterfaceId>,
+    pub imported: bool,
+}
+
+impl Hash for PayloadInfo {
+    fn hash<H: Hasher>(&self, state: &mut H) {
```

Comment:
Mind leaving a comment here explaining why this isn't `derive` and why `self.function` is omitted?